tools/utils/http-helpers.js

```diff
@@ -428,10 +426,9 @@ _.extend(exports, {
         }));
 
       } catch (e) {
-        const size = outputStream.size();
-        const useTry = size === lastSize;
-        const change = size - lastSize;
-        lastSize = outputStream.size();
+        const size = _.reduce(outputStream.body, (s, buf) => s + buf.length, 0);
```

Comment:
We can do a little better than recomputing the total length by keeping a running `sum` variable in the same scope as `outputStream` and only adding the lengths of any new `outputStream.body` elements here. To determine which chunks are new, we can store the previous length of `outputStream.body`.